demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt

```diff
@@ -155,8 +156,7 @@ private fun GravatarTabs(
 private fun ProfileTab(modifier: Modifier = Modifier, onError: (String?, Throwable?) -> Unit) {
     var email by remember { mutableStateOf(BuildConfig.DEMO_EMAIL, neverEqualPolicy()) }
     var hash by remember { mutableStateOf("", neverEqualPolicy()) }
-    var profiles by remember { mutableStateOf(UserProfiles(emptyList()), neverEqualPolicy()) }
-    var loading by remember { mutableStateOf(false) }
+    var profile: UserProfileLoadingState? by remember { mutableStateOf(null, neverEqualPolicy()) }
```

Comment:
Renamed to `profileState`. 

> Another thing if we put all the states in a UserProfileState, we could make profileState non-null since we would always be in an "existing" state (Empty, Loading, Loaded, Error)

I totally agree with this. In fact, that was my original idea when passing a state to the composable. Allow all possible cases. 

The thing is that we don't have all the states right now, so I think we'll be there when we iterate over it with the designs for each state. 

So, I think that is the right approach, but we'll need more iterations until we have all the states within `UserProfileState`.